### DISCUSSION TITLE
Problems doing git pull AGAIN

### DISCUSSION THREAD

**Discussion by rock-crusher** [tagyoureit/nodejs-poolController#748] [Nixie]
After doing a git pull on nodejs-poolController, nodejs-poolController-dashPanel and REM things got screwed up see post:  [https://github.com/tagyoureit/nodejs-poolController/discussions/733](https://github.com/tagyoureit/nodejs-poolController/discussions/733)

Since I am not a brainiac like so many of you here I had no choice but to find the new sd card that I had saved from my first intall of Nixie.

I inserted the new card and everything started up just fine and was workig as it should. Since this was an early version of 7.7.0, I did a pull on pool controller, dash panel and rem.

The below pics show the results, you may need to click on them for a lager view:

![Pool Controller](https://user-images.githubusercontent.com/94593954/235277266-dfd52818-1b27-4c9c-942c-ece921b1b5bd.png)

![Dash Panel](https://user-images.githubusercontent.com/94593954/235277276-557d3ec1-18bb-4026-a408-5ae5dcc158c9.png)

![REM](https://user-images.githubusercontent.com/94593954/235277287-ad47fabc-8473-4322-b8c7-0d9e57f2ef3f.png)

Now when I log on http://raspberrypi:5150 the dash panel is ghosted:

![Dash Panel Ghosted](https://user-images.githubusercontent.com/94593954/235277328-748047a4-ed4b-449a-a6c7-5b76366647f3.png)

When I reboot the pi I get this:

![Unknown Model Not Connected](https://user-images.githubusercontent.com/94593954/235277407-2f49c798-6851-4798-8604-20701f61c1f7.png)

REM seems to load ok and work.

I am now back with problems again.

So, what am I doing wrong and why is this happening for the second time?

**Comment by rstrouse** [comment]
That looks like you have node version 12 installed and you need node version 16+.  Please verify this by running `node -v`

**Comment by rstrouse** [comment]
This looks like the default node version set on your pi is node 12.  To change this run
`nvm install 18`
`nvm use 18`
`nvm alias default 18`

And finally when you run
`node -v`

It should show version 18.x.x

This will show you a list of node versions currently installed
`nvm ls`


After you have set the default node version you will need to run `npm i` in the dashPanel and njsPC directories.

**Reply by rock-crusher** [reply]
Hi rstrouse,

Thanks again for the help.

So I went ahead and did the nvm install 18, the nvm use 18 and nvm alias default 18.

Then finally the node -v. The results are below:

![node -v](https://user-images.githubusercontent.com/94593954/235327890-9ecdc6b9-57d1-42f0-9649-b2e31612421f.png)

![nodejs-poolController npm i](https://user-images.githubusercontent.com/94593954/235327913-00767aac-4ce7-49a3-affd-ec5048e5685e.png)

![nodejs-poolController-dashPanel $ npm i](https://user-images.githubusercontent.com/94593954/235327933-94b4d471-f4ee-4543-aa4a-7c4fcdb34e3c.png)

![npm audit](https://user-images.githubusercontent.com/94593954/235327946-36f9a9cd-0ea9-4d5e-a446-de679d43dad2.png)

![npm i --package-lock-only](https://user-images.githubusercontent.com/94593954/235328190-1caf74e5-94e2-45fe-a8bc-3b9f5778331e.png)

I didn't want to go any further by doing a get pull on nodejs-poolController and nodejs-poolController-dashPanel until you have seen these results.

Please advise my next steps,

Thanks again.

**Reply by rstrouse** [reply]
Ok now delete the node_modules directory from the nodejs-poolController directory .   Then run `npm i` again.  The node_modules contain old dependencies that are not being updated automatically.  

You are golden with dashPanel at this point.  Don't run `npm audit fix` ever.  It often does more damage than fix and you cannot run that in the user root because a package file does not exist there.

**Reply by rock-crusher** [reply]
Ok I deleted the node_modules directory from the nodejs-poolController directory and then ran npm i again on poolController.

The results seem to be the same? 

![nodejs-poolController $ npm i](https://user-images.githubusercontent.com/94593954/235370786-afbad581-efe8-4c8b-8d43-7505ec3f4673.png)

**Reply by CodeSlinger69** [reply]
@rock-crusher Same thing happened to me today as I began experimenting with the 8.0.0 branch. Turns out that the "git pull" command was not completing due to it being unwilling to overwrite the package-lock.json file. This happened in both the pool controller and the dash panel directories.

I did this:
`mv package-lock.json package-lock.json.old`

...then the following worked fine:
`git pull`
`npm i`
